motivaa-backend/motivaa-app/src/test/java/backendtesting/testcases/ExcelOnModulesTest.java

```diff
@@ -1,74 +1,363 @@
+        final String expectedValue = "";
+
+        ProcessObject processObject = RequestMaker.createProcessObject(
+                environmentHost,
+                processStartEndpoint,
+                jSessionId);
 
-        Map<String, Object> requestBody = new HashMap<>();
-        requestBody.put("updateMap", updateMap);
-        requestBody.put("lockVersions", lockVersions);
+        List<OutputVerification> outputVerifications = Arrays.asList(
+                new OutputVerification(excelOutput1Attribute, expectedValue),
+                new OutputVerification(excelOutput2Attribute,expectedValue)
+        );
 
-        String responseFromFirstUpdate = RequestMaker.updateExistingProcessObject(
+        TestUtils.updateProcessObjectAndVerifyOutputValues(
                 environmentHost,
                 processUpdateEndpoint,
-                processUuid,
                 jSessionId,
-                requestBody,
-                "Error when setting " + testedValue + " to " + excelInputAttribute);
+                processObject,
+                excelInputAttribute,
+                inputValue,
+                outputVerifications
+        );
+    }
+    @Test
+    void GIVEN_ExcelAttributeOnParent_HybridInputOutputs_WHEN_second_value_is_set_THEN_both_both_input_values_should_be_copied_to_output() {
+        final String excelInput1Attribute = "ExcelsOnModulesTest.ExcelAttributeOnParent_HybridInputOutputs.input1_shouldNotTriggerExcelUntilAll2InputsHaveValue_notOptional";
+        final String excelInput2Attribute = "ExcelsOnModulesTest.Modules.ExcelsOnModulesTestModule1Link.ExcelsOnModulesTestModule1Link::ExcelsOnModulesTestModule1.ExcelsOnModulesTestModule1Link::ExcelAttributeOnParent_HybridInputOutputs.ExcelsOnModulesTestModule1Link::Input2_shouldTriggerExcel_notOptional";
+        final String input1Value = "noValueShouldTriggerTheExcel";
+        final String input2Value = "excelShouldBeAbleToCopyAnyValue";
+        final String excelOutput1Attribute = "ExcelsOnModulesTest.ExcelAttributeOnParent_HybridInputOutputs.output1_shouldBeSameAsInput1";
+        final String excelOutput2Attribute = "ExcelsOnModulesTest.Modules.ExcelsOnModulesTestModule1Link.ExcelsOnModulesTestModule1Link::ExcelsOnModulesTestModule1.ExcelsOnModulesTestModule1Link::ExcelAttributeOnParent_HybridInputOutputs.ExcelsOnModulesTestModule1Link::output2_shouldBeSameAsInput2";
+        final String expectedValueAfterFirstUpdate = "";
+
+        ProcessObject processObject = RequestMaker.createProcessObject(
+                environmentHost,
+                processStartEndpoint,
+                jSessionId);
+
+        List<OutputVerification> outputVerificationsForFirstUpdate = Arrays.asList(
+                new OutputVerification(excelOutput1Attribute, expectedValueAfterFirstUpdate),
+                new OutputVerification(excelOutput2Attribute,expectedValueAfterFirstUpdate)
+        );
+
+        TestUtils.updateProcessObjectAndVerifyOutputValues(
```

Comment:
maybe you should not assert here, because this is only a setup for the actual testcase if I understand correctly.

Reply:
Maybe the naming is not good enough, this is not when the assertion happens, its where I setup what I want to assert later. Then the assertions happens at updateProcessObjectAndVerifyOutputValues, where this is what I compare the actual values to. 

Thinking about it I think I should rename it to ExpectedValues, which I did now.